src/main/java/br/edu/ifsp/ifspcodelab/gestaoestagiosbackend/campus/CampusRestController.java

```diff
@@ -0,0 +1,50 @@
+package br.edu.ifsp.ifspcodelab.gestaoestagiosbackend.campus;
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("api/v1/campuses")
+@AllArgsConstructor
+public class CampusRestController {
+    private final CampusService campusService;
+
+    @PostMapping
+    public ResponseEntity<CampusDto> create(@Valid @RequestBody CampusCreateDto campusCreateDto) {
```

Comment:
Talvez possamos usar o CampusDto como parâmetro do create e update e tratar na validação caso a caso com groups, da uma olhada na parte final deste artigo https://reflectoring.io/bean-validation-with-spring-boot/. Dessa forma podemos usar um lib de mapeamento pensando em Campus to CampusDto e Campus from CampusDto (vou comentar isso em outra parte da review).

```java
@Value
public class CampusDto {
    @Null(groups = OnCreate.class)
    @NotNull(groups = OnUpdate.class)
    @NotBlank(groups = OnUpdate.class)
    UUID id;
    @NotNull
    @NotBlank
    String name;
    @NotNull
    @NotBlank
    @Size(min = 3, max = 3)
    String abbreviation;
    @NotNull
    @Valid
    AddressDto address;
    @NotNull
    @Valid
    InternshipSectorDto internshipSector;
}
```

No controller:
```java
@PostMapping
@Validated(OnCreate.class)
public ResponseEntity<CampusDto> create(@Valid @RequestBody CampusDto campusDto) {
    Campus campus = campusService.create(campusDto);
    URI uri = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}").buildAndExpand(campus.getId()).toUri();
    return ResponseEntity.created(uri).body(new CampusDto(campus));
}

@PutMapping("{id}")
@Validated(OnUpdate.class)
public ResponseEntity<CampusDto> update(@PathVariable UUID id, @Valid @RequestBody CampusDto campusDto) {
    return ResponseEntity.ok(new CampusDto(campusService.update(id, campusDto)));
}
```